rcl/test/rcl/test_info_by_topic.cpp

```diff
@@ -357,6 +359,7 @@ TEST_F(
     &subscription_options);
   ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
   const std::string fqdn = std::string("/") + this->topic_name;
+  std::this_thread::sleep_for(std::chrono::milliseconds(250));
```

Comment:
If this does help, then I'd recommend polling in a loop, because this only requires a minor hiccup in the machine to still flake.